### DISCUSSION TITLE
Different presets for cooling and heating modes?

### DISCUSSION THREAD

**Discussion by edgimar** [ScratMan/HASmartThermostat#169] [Q&A] [answered]
I have the comfort, sleep, and away temperatures set differently when cooling than when heating.  This means that every time I want to switch from cooling to heating mode or vice versa, I also need to trigger the set-preset-modes service to modify these values.  Is there any way to make this more automatic, or to provide two sets of presets (one for cooling mode, and another for heating mode)?

**Comment by ScratMan** [comment, marked as answer]
Hi, the easiest way is to use an automation in HA, triggered by changing HVAC mode of the thermostat, then add "choose" action based on state of the HVAC mode. If cooling, run an action setting the dedicated temperature in the presets, if heating run similar action with the heating temperature set.
This will work all time, and will not complicate my code more than it is already 👼

**Reply by edgimar** [reply]
That makes sense.  I wasn't aware that you could trigger based on HVAC mode.  Thanks for the idea!
